libs/ui-lib/lib/cim/components/modals/AddHostModal.tsx

```diff
@@ -20,12 +21,24 @@ const AddHostModal: React.FC<AddHostModalProps> = ({
   docVersion,
   isIPXE,
 }) => {
+  const { t } = useTranslation();
   const hasDHCP = infraEnv.metadata?.labels?.networkType !== 'static';
   const sshPublicKey = infraEnv.spec?.sshAuthorizedKey || agentClusterInstall?.spec?.sshPublicKey;
   const { httpProxy, httpsProxy, noProxy } = infraEnv.spec?.proxy || {};
   const imageType = infraEnv.spec?.imageType || 'minimal-iso';
   const [dialogType, setDialogType] = React.useState<AddHostModalStepType>('config');
-  const { t } = useTranslation();
+
+  const [ciscoUrl, setCiscoUrl] = React.useState<string>();
+  const [agentServiceConfig, loaded, error] = useAgentServiceConfig({ name: 'agent' });
+
+  React.useEffect(() => {
+    if (loaded && !error) {
+      if (agentServiceConfig && agentServiceConfig.metadata?.annotations?.['ciscoIntersightURL']) {
+        setCiscoUrl(agentServiceConfig.metadata?.annotations?.['ciscoIntersightURL']);
+      }
+    }
+  }, [agentServiceConfig, error, loaded]);
```

Comment:
no need to store this in state, just calculate it in place
```suggestion
  const [agentServiceConfig, loaded, error] = useAgentServiceConfig({ name: 'agent' });
  
  const ciscoUrl = loaded && !error ? agentServiceConfig?.metadata?.annotations?.['ciscoIntersightURL'] : undefined
```